src/main/java/xyz/nucleoid/plasmid/game/ResourcePackStates.java

```diff
@@ -0,0 +1,85 @@
+package xyz.nucleoid.plasmid.game;
+
+import joptsimple.internal.Strings;
+import net.minecraft.server.network.ServerPlayerEntity;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.Nullable;
+import xyz.nucleoid.plasmid.game.common.GameResourcePack;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+public final class ResourcePackStates {
+    private final Map<UUID, GameResourcePack> resourcePacks = new HashMap<>();
+    private GameResourcePack globalPack = null;
```

Comment:
Hmm, it seems `globalPack` here is the main mechanism for ensuring packs don’t get re-sent between activities- but the handling of resource packs is otherwise per-player. Is it possible to avoid storing a global state like this? Potentially by keeping track of the resource packs per player in the previous activity as well as the new one- the comparison can then be made per-player